### ISSUE TITLE
FFI error from snabb config set with "next-hop {}"

### ISSUE THREAD

**Issue by takikawa** [Igalia/snabbswitch#700]
Supplying `next-hop {}` (which I think is valid according to the schema?) as an argument for `/softwire-config/internal-interface` triggers the FFI error below:

```
$ sudo ./snabb config set -s snabb-softwire-v1 22218 "/softwire-config/internal-interface" "next-hop {}"
lib/yang/data.lua:405: cannot convert 'nil' to 'unsigned int'
stack traceback:
        core/main.lua:149: in function <core/main.lua:147>
        [C]: in function 'finish'
        lib/yang/data.lua:405: in function 'parser'
        program/config/common.lua:77: in function 'parse_command_line'
        program/config/set/set.lua:7: in function 'run'
        program/config/config.lua:23: in function 'run'
        core/main.lua:67: in function <core/main.lua:43>
        [C]: in function 'xpcall'
        core/main.lua:211: in main chunk
        [C]: at 0x004536b0
        [C]: in function 'pcall'
        core/startup.lua:3: in main chunk
        [C]: in function 'require'
        [string "require "core.startup""]:1: in main chunk
```

**Comment by tsyesika** [comment]
My brief look into this, I actually think what you're doing is incorrect based on the schema. My understanding is this (if it were to succeed) would set the contents of `internal-interface` to `next-hop {}`. The `internal-interface` has mandatory fields such as `ip` and `mac`. I think if you wish to set next-hop to empty what you'd do is:

```
$ snabb config set lwaftr-name "/softwire-config/internal-interface/next-hop" "{}"
```

My suspicion as one of the mandatory fields is assumed to be there in the code and so a cast of "nil" to an unsigned int is occurring. I think we probably should check for this on mandatory fields and display a better error.

**Comment by kbara** [comment]
Yes, better error handling around missing mandatory settings with better error messages sounds really good.

**Comment by tsyesika** [comment]
Fixed with PR #730
